Dev/VideoChatClient/AutoInitWinSock/AutoInitWinSock.cpp

```diff
@@ -0,0 +1,25 @@
+#include "AutoInitWinSock.hpp"
+
+AutoInitWinSock::AutoInitWinSock()
+{
+	WSADATA wsaData{};
+	if (0 == WSAStartup(MAKEWORD(2, 2), &wsaData))
+	{
+		throw VideoChatClientException(VideoChatClientStatus::AUTO_INIT_WINSOCK_WSA_STARTUP_FAILED, GetLastError());
```

Comment:
Great Great Great usage of the exception & status classes, this is how I wanted it to be!